### PR TITLE
CI

### PR DESCRIPTION
**Pull request by david-vavra**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      - image: circleci/golang:1.14
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - run:
+          command: |
+            export SLO_EXPORTER_VERSION=${CIRCLE_TAG:-$CIRCLE_SHA1}
+            go get github.com/github-release/github-release
+            make github-release
```
Comment (FUSAKLA) [inline .circleci/config.yml:69]: If the make target has some requirements, it should install it itself ideally

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  version: 2
+  slo-exporter:
+    jobs:
+      - lint
+      - test
+      - build:
+          filters:
+            tags:
+              only: /^v[0-9]+(\.[0-9]+){2}(-.+|[^-.]*)$/
+      - docker_release:
```
Comment (FUSAKLA) [inline .circleci/config.yml]: Maybe we should have separate job for only docker build and run if for every commit and just this release job run for tags?

This way we have no way to verify if the Docker build passes in CI before release.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    docker:
+      - image: circleci/golang:1.14
+    environment:
+      DOCKERHUB_USERNAME=sklikdevops
+    steps:
+      - checkout
+      - setup_remote_docker:
+          version: 17.07.0-ce
+      - attach_workspace:
+          at: .
```
Comment (FUSAKLA) [inline .circleci/config.yml:50]: Not familiar with this, is it necessary?

File: Makefile [hunk 4]
```diff
@@ -1,39 +1,66 @@
...
+.PHONY: docker-build
+docker-build:
+	docker build -t $(DOCKER_IMAGE_REPO):$(SLO_EXPORTER_VERSION) .
+	docker run --rm $(DOCKER_IMAGE_REPO):$(SLO_EXPORTER_VERSION) --help
+
+.PHONY: docker-push
+docker-push:
+	docker push $(DOCKER_IMAGE_REPO):$(SLO_EXPORTER_VERSION)
+
+.PHONY: github-release
```
Comment (FUSAKLA) [inline Makefile:27]: I don't think we need this in Makefile actually, IMHO it's OK to run the script directly in the CI job definition WDYT?

File: Makefile [hunk 5]
```diff
@@ -1,39 +1,66 @@
...
 
+.PHONY: benchmark
 benchmark: clean
 	./scripts/benchmark.sh
 
+.PHONY: test-and-coverage
 test-and-coverage: test
 	go tool cover -func coverage.out
 
+.PHONY: compose
```
Comment (FUSAKLA) [inline Makefile]: I believe this target is deprecated and can be dropped? Event the `clean-compose`.

File: Makefile [hunk 6]
```diff
@@ -1,39 +1,66 @@
...
+.PHONY: github-release
+github-release:
+	bash ci/github_release.sh
+
+.PHONY: lint
 lint:
 	go get github.com/mgechev/revive
 	revive -formatter friendly -config .revive.toml $(shell find $(src_dir) -name "*.go" | grep -v "^$(src_dir)/vendor/")
 
+.PHONY: e2e-test
```
Comment (FUSAKLA) [inline Makefile:37]: We should create an issue to run e2e tests in CI maybe.

File: ci/github_release.sh [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+# extract this particular release from changelog
+awk -v version=${SLO_EXPORTER_VERSION} '$0 ~ /## \[.+\]/ {release = $2} release == "["version"]" {print $0}' CHANGELOG.md > release/CHANGELOG
+
+# build tarballs with built binaries
+for i_file in `find build -name slo_exporter -type f`; do
+    tar -C `dirname $i_file` -czvf `echo $i_file | awk -F/ '{print "release/"$NF"."$(NF-1)".tgz"}'` `basename $i_file`
+done
+
+github-release release \
```
Comment (FUSAKLA) [inline ci/github_release.sh:25]: I wanted to point out the `ghr` tool used in many tutorials but actually found the https://goreleaser.com/quick-start/
which looks pretty neat. It should manage the cross-build, archiving and uploading of the artifacts. 

But if this works for now, I'm pretty much OK with merging this and eventually migrating to the goreleaser in future.

File: CHANGELOG.md [hunk 8]
```diff
@@ -6,19 +6,25 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-# Added
+## Added
 - SLO computation recording rules, alerts
 - slo-exporter grafana dashboard
 
 ## Fixed
 - PrometheusIngester: Fixed unit of the `slo_exporter_prometheus_ingester_query_duration_seconds_bucket` metric.
 
-## [6.0.0] 2020-06-12
+## [v6.0.0] 2020-06-12
```
Comment (FUSAKLA) [inline CHANGELOG.md:27]: Just form curiosity why are we changing these?

**Review by FUSAKLA — APPROVED** [review]
Great! :+1: 
Just few suggestions and questions but I'm OK with merging this even as is.
We can always iterate if needed, for now we need to have the automated test ASAP.

File: CHANGELOG.md (hunk 8 above)
Comment (david-vavra) [reply CHANGELOG.md:27]: I would like to unify format of `git tags`, here mentioned`changelog headers` and docker tags.

File: Makefile (hunk 5 above)
Comment (david-vavra) [reply Makefile]: Yes, those are not used anymore, thanks for pointing this out. :)

Dropped within https://github.com/david-vavra/slo-exporter/pull/1/commits/35d37babe7220506c8ffa2b50bf23f2631d79e50

File: .circleci/config.yml (hunk 1 above)
Comment (david-vavra) [reply .circleci/config.yml:69]: Fixed in https://github.com/david-vavra/slo-exporter/pull/1/commits/35d37babe7220506c8ffa2b50bf23f2631d79e50

File: .circleci/config.yml (hunk 2 above)
Comment (david-vavra) [reply .circleci/config.yml]: Done within https://github.com/david-vavra/slo-exporter/pull/1/commits/9187899ad35e5543554bec397740f49c2d307ffd

File: Makefile (hunk 6 above)
Comment (david-vavra) [reply Makefile:37]: Yeah, I will create it, once this is merged.

File: Makefile (hunk 4 above)
Comment (david-vavra) [reply Makefile:27]: Let's keep it as it is, at least for now.

File: .circleci/config.yml (hunk 3 above)
Comment (david-vavra) [reply .circleci/config.yml:50]: I believe it is. In documentation it is referred to as a mandatory option for [`attach_workspace`](https://circleci.com/docs/2.0/configuration-reference/#attach_workspace)

**Pull request merged**